src/Symfony/Component/Form/Extension/Core/Type/ChoiceType.php

```diff
@@ -60,7 +61,7 @@ public function buildForm(FormBuilderInterface $builder, array $options)
         $choiceList = $this->createChoiceList($options);
         $builder->setAttribute('choice_list', $choiceList);
 
-        if ($options['expanded']) {
+        if ($options['expanded'] && !in_array($options['widget'], array('text', 'hidden'))) {
             $builder->setDataMapper($options['multiple']
```

Comment:
For `checkbox` and `radio`, `multiple` option is ignored, I had made the same suggestion (see linked PR description) and @webmozart preferred to have more explicit names.